### PR TITLE
[SPARK-20713][Spark Core] Convert CommitDenied to TaskKilled.

### PR DESCRIPTION
**Pull request by nlyu**
## What changes were proposed in this pull request?

In executor, toTaskFailedReason is converted to toTaskCommitDeniedReason to avoid the inconsistency of taskState. In JobProgressListener, add case TaskCommitDenied so that now the stage killed number is been incremented other than failed number.
This pull request is picked up from: https://github.com/apache/spark/pull/18070 using commit: ff93ade0248baf3793ab55659042f9d7b8efbdef
The case match for TaskCommitDenied is added incrementing the correct num of killed after pull/18070.

## How was this patch tested?

Run a normal speculative job and check the Stage UI page, should have no failed displayed.

### CODE REVIEW

**Comment by nlyu** [comment]
@tgravescs  Hi Tom i am Louis Lyu, here is the PR

**Comment by tgravescs** [comment]
ok to test

File: core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala [hunk 1]
```diff
@@ -374,6 +374,10 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
         case kill: TaskKilled =>
           execSummary.reasonToNumKilled = execSummary.reasonToNumKilled.updated(
             kill.reason, execSummary.reasonToNumKilled.getOrElse(kill.reason, 0) + 1)
+        case TaskCommitDenied(jobID, partitionID, attemptNumber) =>
```
Comment (tgravescs) [inline core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala]: give this a variable name like case commitDenied: TaskCommitDenied(jobID, partitionID, attemptNumber) that way you can use the variable commitDenied below in the toErrorString

File: core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala [hunk 2]
```diff
@@ -390,6 +394,11 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
             stageData.reasonToNumKilled = stageData.reasonToNumKilled.updated(
               kill.reason, stageData.reasonToNumKilled.getOrElse(kill.reason, 0) + 1)
             Some(kill.toErrorString)
+          case TaskCommitDenied(jobID, partitionID, attemptNumber) =>
```
Comment (tgravescs) [inline core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala]: same here use a variable

File: core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala [hunk 3]
```diff
@@ -428,6 +437,10 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
           case kill: TaskKilled =>
             jobData.reasonToNumKilled = jobData.reasonToNumKilled.updated(
               kill.reason, jobData.reasonToNumKilled.getOrElse(kill.reason, 0) + 1)
+          case TaskCommitDenied(jobID, partitionID, attemptNumber) =>
```
Comment (tgravescs) [inline core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala]: use a variable

**Review by tgravescs — CHANGES_REQUESTED** [review]

File: core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala (hunk 1 above)
Comment (nlyu) [inline core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala]: okay , working on it

**Comment by tgravescs** [comment]
test this please

**Comment by tgravescs** [comment]
+1

**Comment by nlyu** [comment]
@tgravescs Hi Tom a quick question.
So for the UI data table covert JIRA, is there a corresponding JIRA number in the community for that? Like SPARK-20XXX.

**Comment by tgravescs** [comment]
@nlyu  no please file one

**Pull request closed without merging**